### ISSUE TITLE
New icon theme proposal

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#950]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-02-24T06:51:58Z_

This is a proposed patch+assets for a new icon theme based on the Faenza icon theme, which integrates well on mac, linux, and windows.
Screenshot of icon theme attached.

**Attachment:** [spydlib.png](http://code.google.com/p/spyderlib/issues/detail?id=950)

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=950_

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-02-24T05:43:21Z_

Thanks a lot for the improvements Jon. It looks really really good!

Could it be possible to add it as an alternative icon theme? I mean, I would like to maintain the current one (for KDE) and use yours for Mac, Gnome and possibly Windows. How much space does it add to the tree?

**Summary:** New icon theme proposal  
  **Labels:** -Type-Defect Type-Enhancement Cat-SpyderGUI

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-02-24T12:48:18Z_

There wouldn't be a big space problem (a few MB), its just that since i did a lot of re-naming the old theme needs to be adapted to the new naming scheme. On the other hand, i the reason i did re-naming was to use more standard names so it should be quite easy to get icons from the oxygen theme and replace as appropriate. 

Not 100% how do to the icon theme switching. By far the easiest solution would be to ship with this theme and supply alternatives as a zip to overwrite your images directory. I also think that the monochrome icons might still work well in KDE, so maybe it's not to necessary to supply two themes. 

Long story short, I'd like to get this out to people to test - how would i go about submitting my patches and merging with the latest spyder code. At the moment I can merge remote changes into my local copy, but how would i make the modified version availalbe? Is it possible to create a personal branch like in github?

**Comment by spyder-issues-migrator** [comment]
_From [jed.lud...@gmail.com](https://code.google.com/u/114333776555678317161/) on 2012-02-24T14:40:30Z_

Create a personal server-side clone here: https://code.google.com/p/spyderlib/source/clones Create a local clone of this on your machine, and commit your changes there locally. When you are happy with your final merges, push to the server-side clone.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-02-28T01:56:29Z_

Ok, clone created and merge performed. Repository here:

https://jon.chambers3001@code.google.com/r/jonchambers3001-personal/ 

If others would like to test it and eventually merge it into main spyder, that would be awesome !

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-02-28T01:58:24Z_

P.s. Note that this patch also includes a small change to the editor code to remove a special case for mac which disabled "document-style" tabs for the source editor. It seems (on PySide at least) that it works fine.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-03-09T07:52:53Z_

Jon,

Thanks a lot for your contributions. I'll take a look at them during the weekend and please excuse me for my very late answer. I just was fighting a lot to put our new feature "Render math equations" on good grounds.

PS. "document-style" tabs generated a hard crash when the console or edit plugins when detached from the main window. Could you confirm that this is not an issue anymore on Mac?

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-03-09T09:29:25Z_

Ah, i didn't test detaching the editors, it still causes a hard crash, so i've reverted that change in my repo clone.

https://jon.chambers3001@code.google.com/r/jonchambers3001-personal/

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-03-09T09:29:54Z_

Jon, I'm having some free time right now so I'm testing your work. First of all: great work!! I'm really impressed with the new look and I would like it to be on 2.2 for sure! I find it quite pleasing, Spyder looks so more polished and sleek with your work.

Now the minor annoyances:

1. I don't like the folder-open icon. Its brown color doesn't match well with the others. I don't like its form factor either. I think it could be more rectangular. I also don't like set-workingdir, it doesn't feel as sleek as the others.

2. The IPython and Open terminal icons are too alike. It's really hard to tell the difference between them. Perhaps you could take the IP[y] out of the console background to create a different icon? I don't know, it's just a suggestion.

3. Of course, there are some icons that should be replaced because they don't fit anymore. I think before you do it, we couldn't merge your work. It doesn't matter if they come from Faenza or some other icon set, they just have to be monochrome.

Please keep up with your good work and don't get discouraged for our late answers. I'll try hard to be more responsive next time.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-03-09T09:40:41Z_

hey thanks for the feedback. I will try to make the improvements you suggest.

Also, could you point out which icons need to be replaced still. Not all of them need to be 100% monochrome to fit in (e.g. the close buttons will stay in red), but i guess they do need to follow the style.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-03-09T09:41:44Z_

P.s. any idea's for how to translate the "set working directory" metaphor would be very welcome, i struggled with it with is why that icon looks a bit strange...

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-03-09T12:28:52Z_

Ok, more suggestions:

1. Preferences icon should be different from the options one. It should be more like the current one, with some crossed tools. But I was thinking you're right: not every icon has to be monochrome but, perhaps not to disrupt the theme, they should be mostly of one color. So in this case it would be cool to have this icon in blue, with some touches of black, like the "External tools" one. I couldn't find its name but I really like it.

2. "Set working directory" could be a folder with a half-of-its-size arrow in green or red pointing downwards. It's the dir where you hold your data, scripts, plots, etc and so you can simply do

open('a.txt')

to open a file without worrying about the absolute path.

3. When active, next_wng and prev_wng should have its exclamation sign in red, so a user can tell they have really changed its state and he can use them. The same goes to the "Show code analysis/warning errors" and "Show TODO, etc".

4. I would like to see other icons for "Save current file" and "Save all opened files". They are not to clear to me.

5. PYTHONPATH icon is also a bit confusing, because of the red X. I think it was better before, with a plus and minus. You should change that x for a -.

6. I don't like too much the icon associated to Python and html files. I know that those are our current icons but if you could change them it would be great.

7. About the colored icons I would like to see changed, they are mostly 'last_edit_location' and 'prev_cursor' because they are on the main toolbar and (to me) they don't fit well next to 'next_wng' and 'prev_wng'. There is also 'lock_open' (on the object inspector) because it doesn't match 'lock'. And perhaps collapse/expand and collapse_selectin/expand_selection, but they are not so important.

8. 'Re-run last script' and 'Run selection or current block' are not so clear either. Perhaps you should colorize the actions in front of the big play button. I think that could make them better.

I think that's all for now.

**Comment by spyder-issues-migrator** [comment]
_From [jed.lud...@gmail.com](https://code.google.com/u/114333776555678317161/) on 2012-03-09T14:31:16Z_

I hesitate to comment here, but after reading comment 11 I couldn't resist. My goal here is not to discourage but perhaps steer toward something that might be more broadly acceptable.

Large scale user interface changes are almost always controversial. As evidence, see as one example the recent flap over Google Reader: http://groups.google.com/a/googleproductforums.com/forum/#!category-topic/reader/something-is-broken/QxXtCnByeEk The real problem seems to be that user interface "quality" nearly always involves personal preference, particularly when it relates to colors, icons, etc. Once an interface has been out there for a while it becomes very difficult to make changes to it because everyone has adapted their work habits to the old one, and any deviation from that is usually viewed as making things worse simply because it's different.

The way many applications approach this is by skinning. In that light, would it be possible for this new icon set to be available as an optional skin without making it the new default? I'll be honest, I kind of like the look and feel of Spyder as it stands today. Part of that preference is because I've already developed brain synapse connections that respond quickly to the existing icon and color scheme :).

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-03-09T18:35:21Z_

Jed has a good point here. This was also my suggestion on comment 1. I'm almost sure that to get Pierre's approval, the set has to be optional.

So Jon, please rename the old set to match the names of the new one. Then I could code a way to switch between them and to leave yours as default on Mac.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2012-03-09T23:21:05Z_

Following Jed and Carlos last comments, I'd like to mention that I think that changing the icons for better-looking ones is a good thing (great work so far, Jon!) but it has to be as an alternative icon theme. This way, we will make a smooth transition because like Jed wrote above, even if I'm not fully satisfied by Spyder's current look, I'm now used to it (there is a strong mapping in my mind between the icon look and the associated action) and any change will demand an effort to adapt. Of course, that's not criticial for me as I know the software well and I generally like these kind of improvements in other software, but I think that it may be an issue for the average user, slowing down his programming experience with Spyder for a while. 

So +1 for icon themes instead of simply replacing existing icons. 

If you need some ideas to implement the switch between icon themes, I may take a look at it but it seems doable.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-03-10T09:32:02Z_

I tend to be against icon themes, simply because a well designed interface should suit most people without much customization. I also think that jed and pierre overestimate the change this will make to their workflow - given that the icons haven't moved at all, your muscle-memory won't have to adjust, and a well chosen icon will make its purpose instantly clear.

Furthermore, I personally would prefer to spend the (limited) amount of time i have to work on this on making the new icon set as good as it can be.

It would be most helpful if people could try out the new set and see whether the change really hinders their work.

I propose that beta-releases of a new spyder version include only the new icon set so we can get more feedback on it. If most people find the change positive, then for the final release the old icon set can be supplied as a zip-file so that people who really want it can manually overwrite the default version (this is very easy to do). If people are less keen, then it may be worth implementing an in-app icon theme switcher and both themes would be included with the final release.

In any case, I am right now going to focus on improving the new set and so would like to hear comments and suggestions on how it could be improved.

For instance, I would like to know if metaphors used in some of the icons are unclear. I found in creating the new set that icons such as the "settings" icon were used in inappropriate places where they didn't make much sense. I tried to change these but there are definitely still instances of this. I would be glad if people could point them out.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-03-10T09:34:39Z_

P.S I am also against simply adding new features as an option because, as they say "Option menus are the graveyard of good ideas" :P

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2012-03-18T14:09:49Z_

**Labels:** -Type-Enhancement Type-Enh

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-03-18T14:52:24Z_

Ok so here's a small update with a few changed/improved icons. The changes are in my personal repo:
https://jon.chambers3001@code.google.com/r/jonchambers3001-personal/
(note that i merged with dev before pushing, so this is the latest version as of posting + my icons)

**Attachment:** [Screen Shot 2012-03-18 at 10.50.31 PM.png](http://code.google.com/p/spyderlib/issues/detail?id=950#hc18)

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-01-28T16:01:56Z_

**Status:** HelpNeeded  
  **Labels:** MS-v2.3 OpSys-OSX

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-05-11T05:51:04Z_

**Labels:** -MS-v2.3 MS-v2.4

**Comment by spyder-issues-migrator** [comment]
_From [goanp...@gmail.com](https://code.google.com/u/114799823853069473860/) on 2015-01-20T09:08:15Z_

What is the status of this plan? I think that spyder does need an update to its look and a more consistent look and feel

**Comment by flying-sheep** [comment]
i have another proposal: research the names used in common icon themes and use [`QIcon.fromTheme`](http://doc.qt.io/qt-5/qicon.html#fromTheme) instead (example from docs):

```python
undo_icon = QIcon.fromTheme('edit-undo', QIcon(':/undo.png'));
```

as you can see, you can fallback to provided icons, but we can use those delicious breeze icons on Plasma 5.

**Comment by ccordoba12** [comment]
@flying-sheep that's a good idea but it's not multiplatform :-)

Right now @SylvainCorlay is working on creating a theme based on Font Awesome.

**Comment by flying-sheep** [comment]
> it's not multiplatform

sure is, it’ll fall back to the second argument.

also it will try the [themePaths](http://doc.qt.io/qt-5/qicon.html#themeSearchPaths), which one can use to supply a freedesktop-compatible icon theme for all platforms.

**Comment by ccordoba12** [comment]
I'd really prefer that Spyder looked the same in all platforms (I, for 
one, move between all platforms all the time :-)

**Comment by SylvainCorlay** [comment]
@flying-sheep I am looking at the `QIcon.fromTheme` API which is quite nice. Even if we don't use it *per se*, we could follow that Qt idiom to handle themes, or even third-party themes and fallback icons. 

How about we continue this discussion in Issue #2210 ? (will close this issue if everyone is ok.)

**Comment by flying-sheep** [comment]
ok, let’s discuss elsewhere
